plugins/inputs/sqlserver/azuresqlqueries.go

```diff
@@ -382,6 +382,116 @@ HAVING
 OPTION(RECOMPILE);
 `
 
+const sqlAzureDBPartQueryPeriod = `
+DECLARE @QueryData NVARCHAR(MAX) = ?1; --input parameter
+
+DECLARE @lastIntervalEndTimestamp BIGINT = 0;
+
+IF @QueryData != '' AND @QueryData IS NOT NULL
+	SET @lastIntervalEndTimestamp = CAST(@QueryData AS BIGINT);
+
+DECLARE @magicNumberDays BIGINT = 4294967296, @magicNumberSeconds BIGINT = 300, @secondsInDay INT = 86400;
+
+DECLARE @timeGrain BIGINT = (SELECT interval_length_minutes * 60 FROM sys.database_query_store_options WITH (NOLOCK)); --use query store time grain
+
+DECLARE @currDate DATETIME = GETUTCDATE();
+DECLARE @currIntervalEndTimestamp BIGINT = CONVERT(BIGINT, DATEDIFF(day, 0, @currDate)) * @secondsInDay + DATEDIFF(second, DATEADD(day, DATEDIFF(day, 0, @currDate), 0), @currDate) / @timeGrain * @timeGrain;
+IF @lastIntervalEndTimestamp = @currIntervalEndTimestamp
+	RETURN;
+
+DECLARE @currIntervalStartTimestamp BIGINT = @currIntervalEndTimestamp - @timeGrain;
+IF @lastIntervalEndTimestamp != 0
+	SET @currIntervalStartTimestamp = @lastIntervalEndTimestamp;
+
+DECLARE @currIntervalEndDate datetime = DATEADD(second, @currIntervalEndTimestamp % @secondsInDay, DATEADD(day, @currIntervalEndTimestamp / @secondsInDay, 0));
+DECLARE @currIntervalStartDate datetime = DATEADD(second, @currIntervalStartTimestamp % @secondsInDay, DATEADD(day, @currIntervalStartTimestamp / @secondsInDay, 0));
+
+`
+
+const sqlAzureDBQueryStoreRuntimeStatistics = sqlAzureDBPartQueryPeriod + `
+SET DEADLOCK_PRIORITY -10;
+IF SERVERPROPERTY('EngineEdition') = 5  -- Is this Azure SQL DB?
+BEGIN
+SELECT
+	'sqlserver_azuredb_querystore_runtime_stats' AS [measurement],
+	CONVERT(varchar(36), NEWID()) AS [measurement_id],
```

Comment:
Removed. Measurement_ID was needed because in the output recordset we had multiple records for the same query in the same time interval, but it is no longer needed now that aggregation is added https://github.com/influxdata/telegraf/pull/8465/commits/3cf25f6ccab56ae377db346adef5d42c60ed7d70